### ISSUE TITLE
Randomly Shuffling Access Ordering of Graphs

### ISSUE THREAD

**Issue by edogawashinichi** [ZigRazor/CXXGraph#439]
**Is your feature request related to a problem? Please describe.**
[issue433](https://github.com/ZigRazor/CXXGraph/issues/433)
[issue408](https://github.com/ZigRazor/CXXGraph/issues/408)
Bug-free code is rare in medium and large projects. 
Therefore, test coverage is extremely important. 
Due to differences in operating systems, compilers, etc., 
the access order of graphs in test cases may also differ, 
which may cause test cases to behave differently. 

**Describe the solution you'd like**
By randomly shuffling the access order of the graph, 
we can easily obtain a large number of equivalent test cases, 
allowing us to find hidden bugs as much as possible. 
So I propose to add a test class between the test case and the graph class to allow random access.

**Describe alternatives you've considered**
Open a test branch，and add this feature to it.

**Additional context**

**Comment by edogawashinichi** [comment]
I have implemented the alternative locally. 
You do have the opportunity to find more bugs each time you run the test cases.
<img width="818" alt="image" src="https://github.com/ZigRazor/CXXGraph/assets/37181294/87eaf78c-177c-48f2-b23c-b1c1f2b9b62f">

**Comment by edogawashinichi** [comment]
and more bugs I've found
<img width="817" alt="image" src="https://github.com/ZigRazor/CXXGraph/assets/37181294/fdf1a2f1-19f7-4316-85b8-46a5ef947cc3">

**Comment by nolankramer** [comment]
This is a great idea - as long as we can always reproduce test cases by using the same seed. Fuzzing is a great way to ensure robust coverage and functionality.

**Comment by edogawashinichi** [comment]
> This is a great idea - as long as we can always reproduce test cases by using the same seed. Fuzzing is a great way to ensure robust coverage and functionality.

Yep, technically using the same seed is a precise and efficient way.
Currently the method I use is to save the vertex traversal ordering of the buggy test case.

**Comment by ZigRazor** [comment]
Good Idea!

**Comment by badumbatish** [comment]
@edogawashinichi this must be what you're referring to in #433 

I hope it's not too much to ask but would you please create a draft PR? would love to learn how to implement this

**Comment by edogawashinichi** [comment]
> @edogawashinichi this must be what you're referring to in #433
> 
> I hope it's not too much to ask but would you please create a draft PR? would love to learn how to implement this

Sure, I just implement a very rough test branch that help trace bugs caused by ordering.
<img width="819" alt="image" src="https://github.com/ZigRazor/CXXGraph/assets/37181294/d3cfd293-f3c8-4a5b-ab98-984110ae08fd">
